### DISCUSSION TITLE
Why is event input of type std::string and not char?

### DISCUSSION THREAD

**Discussion by ghost** [ArthurSonzogni/FTXUI#374] [Q&A] [answered]
I was wondering why the `input_` variable in `Event` struct is of type `std::string` and not `char`. To me it looks like it only stores one character for each key press or does it store more than one char for some event?

**Comment by ArthurSonzogni** [comment, marked as answer]
UTF8 characters can take up to 4 char.
Also, an event can be some mouse event, which can take 11 char.

You can try the example:
https://arthursonzogni.com/FTXUI/examples/?file=component/print_key_press

It prints on the left, the array of char the terminal sent us.
